### PR TITLE
 python311Packages.cherrypy: 18.8.0 -> 18.9.0 

### PR DESCRIPTION
**Pull request by fabaff**
Changelog: https://github.com/cherrypy/cherrypy/blob/v18.9.0/CHANGES.rst

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 291170` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>35 packages failed to build:</summary>
  <ul>
    <li>ceph (ceph-dev)</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>ceph.dev (ceph-dev.dev)</li>
    <li>ceph.doc (ceph-dev.doc)</li>
    <li>libceph (ceph.lib ,libceph.dev ,libceph.doc ,libceph.lib ,libceph.man)</li>
    <li>ceph.man (ceph-dev.man)</li>
    <li>python311Packages.mapbox</li>
    <li>python311Packages.mapbox.dist</li>
    <li>python312Packages.cachecontrol</li>
    <li>python312Packages.cachecontrol.dist</li>
    <li>python312Packages.cherrypy</li>
    <li>python312Packages.cherrypy-cors</li>
    <li>python312Packages.cherrypy-cors.dist</li>
    <li>python312Packages.cherrypy.dist</li>
    <li>python312Packages.cwl-upgrader</li>
    <li>python312Packages.cwl-upgrader.dist</li>
    <li>python312Packages.cwl-utils</li>
    <li>python312Packages.cwl-utils.dist</li>
    <li>python312Packages.mapbox</li>
    <li>python312Packages.mapbox.dist</li>
    <li>python312Packages.pylxd</li>
    <li>python312Packages.pylxd.dist</li>
    <li>python312Packages.schema-salad</li>
    <li>python312Packages.schema-salad.dist</li>
    <li>python312Packages.ws4py</li>
    <li>python312Packages.ws4py.dist</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.ga</li>
    <li>sambaFull (samba4Full)</li>
    <li>sambaFull.dev (samba4Full.dev)</li>
    <li>sambaFull.man (samba4Full.man)</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>95 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>asciidoc-full-with-plugins.dist</li>
    <li>asciidoc-full.dist</li>
    <li>ayu-theme-gtk</li>
    <li>breeze-hacked-cursor-theme</li>
    <li>capitaine-cursors</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>cinnamon.iso-flags-png-320x420</li>
    <li>cinnamon.iso-flags-svg</li>
    <li>clevis</li>
    <li>clevis.man</li>
    <li>coreboot-configurator</li>
    <li>cwltool</li>
    <li>cwltool.dist</li>
    <li>dblatexFull</li>
    <li>disorderfs</li>
    <li>emojione</li>
    <li>fim</li>
    <li>flexget</li>
    <li>flexget.dist</li>
    <li>frigate</li>
    <li>hackneyed</li>
    <li>hikounomizu</li>
    <li>inkscape</li>
    <li>inkscape-extensions.textext</li>
    <li>inkscape-extensions.textext.dist</li>
    <li>inkscape-with-extensions</li>
    <li>iso-flags</li>
    <li>k40-whisperer</li>
    <li>kabeljau</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>luksmeta</li>
    <li>mate.mate-utils</li>
    <li>material-cursors</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mojave-gtk-theme</li>
    <li>numix-cursor-theme</li>
    <li>numix-solarized-gtk-theme</li>
    <li>pdm</li>
    <li>pdm.dist</li>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
    <li>plata-theme</li>
    <li>poetry</li>
    <li>poetry.dist</li>
    <li>poetryPlugins.poetry-audit-plugin</li>
    <li>poetryPlugins.poetry-audit-plugin.dist</li>
    <li>poetryPlugins.poetry-plugin-up</li>
    <li>poetryPlugins.poetry-plugin-up.dist</li>
    <li>pop-gtk-theme</li>
    <li>python311Packages.cachecontrol</li>
    <li>python311Packages.cachecontrol.dist</li>
    <li>python311Packages.cherrypy</li>
    <li>python311Packages.cherrypy-cors</li>
    <li>python311Packages.cherrypy-cors.dist</li>
    <li>python311Packages.cherrypy.dist</li>
    <li>python311Packages.cwl-upgrader</li>
    <li>python311Packages.cwl-upgrader.dist</li>
    <li>python311Packages.cwl-utils</li>
    <li>python311Packages.cwl-utils.dist</li>
    <li>python311Packages.diagrams</li>
    <li>python311Packages.diagrams.dist</li>
    <li>python311Packages.orange-canvas-core</li>
    <li>python311Packages.orange-canvas-core.dist</li>
    <li>python311Packages.orange-widget-base</li>
    <li>python311Packages.orange-widget-base.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python311Packages.pyalgotrade</li>
    <li>python311Packages.pyalgotrade.dist</li>
    <li>python311Packages.pylxd</li>
    <li>python311Packages.pylxd.dist</li>
    <li>python311Packages.schema-salad</li>
    <li>python311Packages.schema-salad.dist</li>
    <li>python311Packages.scikit-bio</li>
    <li>python311Packages.scikit-bio.dist</li>
    <li>python311Packages.slack-bolt</li>
    <li>python311Packages.slack-bolt.dist</li>
    <li>python311Packages.ws4py</li>
    <li>python311Packages.ws4py.dist</li>
    <li>python312Packages.diagrams</li>
    <li>python312Packages.diagrams.dist</li>
    <li>sabnzbd</li>
    <li>tang</li>
    <li>tang.man</li>
    <li>tebreak</li>
    <li>tebreak.dist</li>
    <li>vimix-cursors</li>
    <li>volantes-cursors</li>
    <li>wpa_supplicant_gui</li>
  </ul>
</details>

File: pkgs/development/python-modules/cherrypy/default.nix [hunk 1]
```diff
@@ -17,43 +18,56 @@
...
-      --replace "-p pytest_cov" "" \
-      --replace "--no-cov-on-fail" ""
+      --replace-fail "--doctest-modules" "-vvv" \
+      --replace-fail "-p pytest_cov" "" \
+      --replace-fail "--no-cov-on-fail" ""
     sed -i "/--cov/d" pytest.ini
   '';
 
   nativeBuildInputs = [
+    setuptools
```
Comment (dotlambda) [inline pkgs/development/python-modules/cherrypy/default.nix:61]: not needed

File: pkgs/development/python-modules/cherrypy/default.nix [hunk 2]
```diff
@@ -115,17 +129,26 @@ buildPythonPackage rec {
...
+    ];
+    memcached_session = [
+      python-memcached
+    ];
+    routes_dispatcher = [
+      routes
+    ];
+    ssl = [
+      pyopenssl
+    ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/cherrypy/default.nix:143]: Let's keep the diff small.

File: pkgs/development/python-modules/pytest-services/default.nix [hunk 3]
```diff
@@ -1,46 +1,79 @@
...
+
   nativeBuildInputs = [
     setuptools-scm
     toml
   ];
 
-  buildInputs = [ pytest ];
+  buildInputs = [
+    pytest
+  ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-services/default.nix:48]: Same here

File: pkgs/development/python-modules/pylxd/default.nix [hunk 4]
```diff
@@ -1,21 +1,27 @@
...
 buildPythonPackage rec {
   pname = "pylxd";
   version = "2.3.2";
-  format = "setuptools";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
     owner = "lxc";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pylxd/default.nix:27]:
```suggestion
    owner = "canonical";
```

**Review by dotlambda — COMMENTED** [review]
Typo: "relace"

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 291170` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.mapbox</li>
    <li>python311Packages.mapbox.dist</li>
    <li>python312Packages.cachecontrol</li>
    <li>python312Packages.cachecontrol.dist</li>
    <li>python312Packages.cherrypy</li>
    <li>python312Packages.cherrypy-cors</li>
    <li>python312Packages.cherrypy-cors.dist</li>
    <li>python312Packages.cherrypy.dist</li>
    <li>python312Packages.cwl-upgrader</li>
    <li>python312Packages.cwl-upgrader.dist</li>
    <li>python312Packages.cwl-utils</li>
    <li>python312Packages.cwl-utils.dist</li>
    <li>python312Packages.mapbox</li>
    <li>python312Packages.mapbox.dist</li>
    <li>python312Packages.pylxd</li>
    <li>python312Packages.pylxd.dist</li>
    <li>python312Packages.schema-salad</li>
    <li>python312Packages.schema-salad.dist</li>
    <li>python312Packages.slack-bolt</li>
    <li>python312Packages.slack-bolt.dist</li>
    <li>python312Packages.ws4py</li>
    <li>python312Packages.ws4py.dist</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>14 packages failed to build:</summary>
  <ul>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>ceph.dev</li>
    <li>ceph.doc</li>
    <li>libceph (ceph.lib ,libceph.dev ,libceph.doc ,libceph.lib ,libceph.man)</li>
    <li>ceph.man</li>
    <li>frigate</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.ga</li>
    <li>samba4Full</li>
    <li>samba4Full.dev</li>
    <li>samba4Full.man</li>
  </ul>
</details>
<details>
  <summary>98 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>asciidoc-full-with-plugins.dist</li>
    <li>asciidoc-full.dist</li>
    <li>ayu-theme-gtk</li>
    <li>breeze-hacked-cursor-theme</li>
    <li>capitaine-cursors</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>cinnamon.iso-flags-png-320x420</li>
    <li>cinnamon.iso-flags-svg</li>
    <li>clevis</li>
    <li>clevis.man</li>
    <li>coreboot-configurator</li>
    <li>cwltool</li>
    <li>cwltool.dist</li>
    <li>dblatexFull</li>
    <li>disorderfs</li>
    <li>emojione</li>
    <li>fim</li>
    <li>flexget</li>
    <li>flexget.dist</li>
    <li>hackneyed</li>
    <li>hikounomizu</li>
    <li>inkscape</li>
    <li>inkscape-extensions.textext</li>
    <li>inkscape-extensions.textext.dist</li>
    <li>inkscape-with-extensions</li>
    <li>iso-flags</li>
    <li>k40-whisperer</li>
    <li>kabeljau</li>
    <li>kakounePlugins.rep</li>
    <li>luksmeta</li>
    <li>mate.mate-utils</li>
    <li>material-cursors</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mojave-gtk-theme</li>
    <li>numix-cursor-theme</li>
    <li>numix-solarized-gtk-theme</li>
    <li>pdm</li>
    <li>pdm.dist</li>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
    <li>plata-theme</li>
    <li>poetry</li>
    <li>poetry.dist</li>
    <li>poetryPlugins.poetry-audit-plugin</li>
    <li>poetryPlugins.poetry-audit-plugin.dist</li>
    <li>poetryPlugins.poetry-plugin-up</li>
    <li>poetryPlugins.poetry-plugin-up.dist</li>
    <li>pop-gtk-theme</li>
    <li>python311Packages.cachecontrol</li>
    <li>python311Packages.cachecontrol.dist</li>
    <li>python311Packages.cherrypy</li>
    <li>python311Packages.cherrypy-cors</li>
    <li>python311Packages.cherrypy-cors.dist</li>
    <li>python311Packages.cherrypy.dist</li>
    <li>python311Packages.cwl-upgrader</li>
    <li>python311Packages.cwl-upgrader.dist</li>
    <li>python311Packages.cwl-utils</li>
    <li>python311Packages.cwl-utils.dist</li>
    <li>python311Packages.diagrams</li>
    <li>python311Packages.diagrams.dist</li>
    <li>python311Packages.orange-canvas-core</li>
    <li>python311Packages.orange-canvas-core.dist</li>
    <li>python311Packages.orange-widget-base</li>
    <li>python311Packages.orange-widget-base.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python311Packages.pyalgotrade</li>
    <li>python311Packages.pyalgotrade.dist</li>
    <li>python311Packages.pylxd</li>
    <li>python311Packages.pylxd.dist</li>
    <li>python311Packages.pytest-services</li>
    <li>python311Packages.pytest-services.dist</li>
    <li>python311Packages.schema-salad</li>
    <li>python311Packages.schema-salad.dist</li>
    <li>python311Packages.scikit-bio</li>
    <li>python311Packages.scikit-bio.dist</li>
    <li>python311Packages.slack-bolt</li>
    <li>python311Packages.slack-bolt.dist</li>
    <li>python311Packages.ws4py</li>
    <li>python311Packages.ws4py.dist</li>
    <li>python312Packages.diagrams</li>
    <li>python312Packages.diagrams.dist</li>
    <li>python312Packages.pytest-services</li>
    <li>python312Packages.pytest-services.dist</li>
    <li>sabnzbd</li>
    <li>tang</li>
    <li>tang.man</li>
    <li>tebreak</li>
    <li>tebreak.dist</li>
    <li>vimix-cursors</li>
    <li>volantes-cursors</li>
    <li>wpa_supplicant_gui</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 291170` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.mapbox</li>
    <li>python311Packages.mapbox.dist</li>
    <li>python312Packages.mapbox</li>
    <li>python312Packages.mapbox.dist</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>cwltool</li>
    <li>cwltool.dist</li>
    <li>python312Packages.slack-bolt</li>
    <li>python312Packages.slack-bolt.dist</li>
  </ul>
</details>
<details>
  <summary>130 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>asciidoc-full-with-plugins.dist</li>
    <li>asciidoc-full.dist</li>
    <li>ayu-theme-gtk</li>
    <li>breeze-hacked-cursor-theme</li>
    <li>capitaine-cursors</li>
    <li>ceph (ceph-dev)</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>ceph.dev (ceph-dev.dev)</li>
    <li>ceph.doc (ceph-dev.doc)</li>
    <li>libceph (ceph.lib ,libceph.dev ,libceph.doc ,libceph.lib ,libceph.man)</li>
    <li>ceph.man (ceph-dev.man)</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>cinnamon.iso-flags-png-320x420</li>
    <li>cinnamon.iso-flags-svg</li>
    <li>clevis</li>
    <li>clevis.man</li>
    <li>coreboot-configurator</li>
    <li>dblatexFull</li>
    <li>disorderfs</li>
    <li>emojione</li>
    <li>fim</li>
    <li>flexget</li>
    <li>flexget.dist</li>
    <li>frigate</li>
    <li>hackneyed</li>
    <li>hikounomizu</li>
    <li>inkscape</li>
    <li>inkscape-extensions.textext</li>
    <li>inkscape-extensions.textext.dist</li>
    <li>inkscape-with-extensions</li>
    <li>iso-flags</li>
    <li>k40-whisperer</li>
    <li>kabeljau</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>luksmeta</li>
    <li>mate.mate-utils</li>
    <li>material-cursors</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mojave-gtk-theme</li>
    <li>numix-cursor-theme</li>
    <li>numix-solarized-gtk-theme</li>
    <li>pdm</li>
    <li>pdm.dist</li>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
    <li>plata-theme</li>
    <li>poetry</li>
    <li>poetry.dist</li>
    <li>poetryPlugins.poetry-audit-plugin</li>
    <li>poetryPlugins.poetry-audit-plugin.dist</li>
    <li>poetryPlugins.poetry-plugin-up</li>
    <li>poetryPlugins.poetry-plugin-up.dist</li>
    <li>pop-gtk-theme</li>
    <li>python311Packages.cachecontrol</li>
    <li>python311Packages.cachecontrol.dist</li>
    <li>python311Packages.cherrypy</li>
    <li>python311Packages.cherrypy-cors</li>
    <li>python311Packages.cherrypy-cors.dist</li>
    <li>python311Packages.cherrypy.dist</li>
    <li>python311Packages.cwl-upgrader</li>
    <li>python311Packages.cwl-upgrader.dist</li>
    <li>python311Packages.cwl-utils</li>
    <li>python311Packages.cwl-utils.dist</li>
    <li>python311Packages.diagrams</li>
    <li>python311Packages.diagrams.dist</li>
    <li>python311Packages.orange-canvas-core</li>
    <li>python311Packages.orange-canvas-core.dist</li>
    <li>python311Packages.orange-widget-base</li>
    <li>python311Packages.orange-widget-base.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python311Packages.osc-diagram</li>
    <li>python311Packages.osc-diagram.dist</li>
    <li>python311Packages.pyalgotrade</li>
    <li>python311Packages.pyalgotrade.dist</li>
    <li>python311Packages.pylxd</li>
    <li>python311Packages.pylxd.dist</li>
    <li>python311Packages.pytest-services</li>
    <li>python311Packages.pytest-services.dist</li>
    <li>python311Packages.schema-salad</li>
    <li>python311Packages.schema-salad.dist</li>
    <li>python311Packages.scikit-bio</li>
    <li>python311Packages.scikit-bio.dist</li>
    <li>python311Packages.slack-bolt</li>
    <li>python311Packages.slack-bolt.dist</li>
    <li>python311Packages.ws4py</li>
    <li>python311Packages.ws4py.dist</li>
    <li>python312Packages.cachecontrol</li>
    <li>python312Packages.cachecontrol.dist</li>
    <li>python312Packages.cherrypy</li>
    <li>python312Packages.cherrypy-cors</li>
    <li>python312Packages.cherrypy-cors.dist</li>
    <li>python312Packages.cherrypy.dist</li>
    <li>python312Packages.cwl-upgrader</li>
    <li>python312Packages.cwl-upgrader.dist</li>
    <li>python312Packages.cwl-utils</li>
    <li>python312Packages.cwl-utils.dist</li>
    <li>python312Packages.diagrams</li>
    <li>python312Packages.diagrams.dist</li>
    <li>python312Packages.osc-diagram</li>
    <li>python312Packages.osc-diagram.dist</li>
    <li>python312Packages.pylxd</li>
    <li>python312Packages.pylxd.dist</li>
    <li>python312Packages.pytest-services</li>
    <li>python312Packages.pytest-services.dist</li>
    <li>python312Packages.schema-salad</li>
    <li>python312Packages.schema-salad.dist</li>
    <li>python312Packages.ws4py</li>
    <li>python312Packages.ws4py.dist</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.ga</li>
    <li>sabnzbd</li>
    <li>sambaFull (samba4Full)</li>
    <li>sambaFull.dev (samba4Full.dev)</li>
    <li>sambaFull.man (samba4Full.man)</li>
    <li>tang</li>
    <li>tang.man</li>
    <li>tebreak</li>
    <li>tebreak.dist</li>
    <li>vimix-cursors</li>
    <li>volantes-cursors</li>
    <li>wpa_supplicant_gui</li>
  </ul>
</details>

**Comment by fabaff** [comment]
`falcon` and `cwltool` are already broken.

**Pull request merged**